amazon-kinesis-client/src/main/java/software/amazon/kinesis/common/StreamARNUtil.java

```diff
@@ -0,0 +1,86 @@
+package software.amazon.kinesis.common;
+
+import com.google.common.base.Joiner;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.model.GetCallerIdentityResponse;
+
+import java.util.HashMap;
+import java.util.Optional;
+
+@Slf4j
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class StreamARNUtil {
+    private static final HashMap<String, Arn> streamARNCache = new HashMap<>();
```

Comment:
FYI: I have submitted https://github.com/awslabs/amazon-kinesis-client/pull/1096 to provide a `SupplierCache`. This may be leveraged here to provide a singular `Arn` cache. Example:

```
private static final SupplierCache<Arn> CACHED_ARN = new SupplierCache<>(() -> {
        try {
            final GetCallerIdentityResponse response = getStsClient().getCallerIdentity();
            return Arn.fromString(response.arn());
        } catch (AwsServiceException | SdkClientException e) {
            log.warn("Unable to get sts caller identity to build stream arn", e);
            return null;
        }
    });
```